### DISCUSSION TITLE
Accessing Shortening service without API?

### DISCUSSION THREAD

**Discussion by sascha-hendel** [YOURLS/YOURLS#3691] [Q&A] [answered]
Hi,
is it possible to get shortened urls directly by including PHP files from YOURLS into the PHP code from the requesting website? I try to avoid API use with web protocol. YOURLS and "master" website are on the same server.

**Comment by dgw** [comment]
Maybe, but it wouldn't be a supported use.

**Reply by sascha-hendel** [reply]
Ok, I searched a litle bit and found some hints in the plugin section.
With `require_once('PATH/TO/load-yourls.php');` and then calling the function `$result = yourls_add_new_link( $url, $keyword, $title );` a result array with the shortened url is delivered. This seems to work for me.
Does something  speaks against that use (except that it is not "officially" supported)?

**Comment by ozh** [comment, marked as answer]
Indeed, you can load `load-yourls.php` on your own.
Also see included file `sample-remote-api-call.txt` at the root for inspiration. As @dgw said, this is provided as an example but probably not supported :)

**Reply by sascha-hendel** [reply]
I already had a look on `sample-remote-api-call.txt` ;) .But this example creates much overhead with CURL requests.
I would really like to use the direct way with  `$result = yourls_add_new_link( $url, $keyword, $title );` . But maybe there are some pitfalls (for example potential name space conflicts,...). Would be helpful, if experienced users post their opinion and hints here. Can you therefore please mark this question as OPEN again for a while?

**Comment by sascha-hendel** [comment]
I list here now my own experiences (for other folks who want to use YOURLS in the same way):

- Be aware: YOURLS uses the superglobal $_REQUEST . Own settings of this variable will be reset on invoking `require_once('PATH/TO/load-yourls.php');`
- Recommended: Disable flood check for (only direct) requests to avoid errors, place something like `if($localrequest){define('YOURLS_FLOOD_DELAY_SECONDS', 0 );}` where variable localrequest must be true on invoking `require_once('PATH/TO/load-yourls.php');`
- Be aware: In direct request mode the superglobal `$_SERVER['HTTP_HOST']` isn't set! If You have a multiple domain setup (and use this variable for selecting the proper settings in your yourls config.php).
- **UNSOLVED PROBLEM**: If a (database) error occurs YOURLS echos a HTML page with error information. In this case this is not useful. Is there a way to catch error messages?
